lib/ansible/modules/messaging/kafka_topics.py

```diff
@@ -0,0 +1,338 @@
+
+    def create(self):
+        ''' Create new kafka topic. '''
+
+        # create action supports only 1 topic
+        if len(self.topic) > 1:
+                msg = 'create action supports only 1 topic, %s provided' % (len(self.topic))
+                return self.module.fail_json(msg)
+        else:
+            if self.config:
+                cmd = ('%s --create --if-not-exists --zookeeper %s '
+                       '--replication-factor %s --partitions %s '
+                       '--topic %s %s') % (self.executable,
+                                           self.zookeeper,
+                                           self.replication_factor,
+                                           self.partitions,
+                                           ''.join(self.topic),
+                                           self.config)
+
+            else:
+                cmd = ('%s --create --if-not-exists --zookeeper %s '
+                       '--replication-factor %s --partitions %s '
+                       '--topic %s') % (self.executable,
+                                        self.zookeeper,
+                                        self.replication_factor,
+                                        self.partitions,
+                                        ''.join(self.topic))
+
+        return self.module.run_command(cmd)
+
+    def delete(self):
+        ''' Delete a kafka topic. '''
+
+        # delete action supports only 1 topic
+        if len(self.topic) > 1:
+            msg = 'delete action supports only 1 topic, %s provided' % (len(self.topic))
+            return self.module.fail_json(msg)
+        else:
+            cmd = ('%s --delete --zookeeper %s '
+                   '--topic %s --if-exists') % (self.executable,
+                                                self.zookeeper,
+                                                self.topic[0])
+
+        return self.module.run_command(cmd)
+
+
+def main():
+    argument_spec = dict(
+        action=dict(default='list',
+                    choices=['list', 'describe', 'alter',
+                             'create', 'delete'],
+                    type='str'),
+        topic=dict(type='list'),
+        config=dict(type='dict'),
+        zookeeper=dict(required=True, type='list'),
+        replication_factor=dict(type='int'),
+        partitions=dict(type='int'),
+        executable=dict(default='/usr/bin/kafka-topics',
+                        type='path'),
+    )
+
+    required_if = [
+        ['action', 'alter', ['topic', 'partitions']],
+        ['action', 'create',
+         ['topic', 'replication_factor', 'partitions']
+         ],
+        ['action', 'delete', ['topic']],
+    ]
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_if=required_if,
+        supports_check_mode=False,
+    )
+
+    changed = False
+    failed = False
+
+    if module.params['executable']:
+        if not os.path.isfile(module.params['executable']):
```

Comment:
`module.param['executable']` could be a non executable file. [`ansible.module_utils.basic.is_executable`](https://github.com/ansible/ansible/blob/68aeaa5/lib/ansible/module_utils/basic.py#L657) should be used in order to check that.